docs/pages/why-web-components.md

```diff
@@ -0,0 +1,124 @@
+- The <a href="https://designsystem.digital.gov/">U.S. Web Design System</a>
+- Other systems also built with web components, like <a href="https://opensource.adobe.com/spectrum-web-components/">Adobe&rsquo;s Spectrum</a>
+- Systems with fantastic documentation for new contributors, like&nbsp;<a href="https://opensource.adobe.com/spectrum-web-components/">Clarity</a>
+
+### Components encapsulate code
+
+The code for each widget is within the custom element class. Our component patterns with local variables protect us from accidentally influencing global variables. The custom element API provides a clear interface for interacting with the rest of the application. It receives events responding to instantiation, removal, and attribute updates.
+
+## When we use web components
+
+We only need web components when we need JavaScript. We do not always need JavaScript. If HTML and CSS provide the desired features, that&rsquo;s all a component uses.
+
+## Progressive enhancement
+
+We cannot depend&nbsp;on every device supporting our web components, even if they work in the latest version of all browsers. Older devices get locked out of update cycles. Some people avoid freely available software updates. 
+
+Critical components should work even if the JavaScript fails. For example, accordions have always been progressively enhanced. Now they use details and summary elements. This lets people read the content, even if the JavaScript fails to execute. This failure state eliminates the extra styling or open/close animations.
+
+If a component is not critical to a site&rsquo;s mission, we skip progressive enhancement. We have not made the <a href="https://designsystem.webstandards.ca.gov/components/page-feedback/readme/">page feedback</a> component progressively enhanced because it&#39;s not why people visit the page. That element is only visible when the visitor&rsquo;s browser successfully executes its code. We also use script type=module, which is ignored by browsers that do not support ES6 components. In these cases, the network request is not even made. This stops the JavaScript from slowing down the basic experience.
+
+## Features of web components we use
+
+Web components help limit development to the local scope of the custom element. There are good <a href="https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_custom_elements">lifecycle tools</a> for building up and tearing down JavaScript features. In other words, no waiting for &quot;DOMContentLoaded&quot; or any other old hacks.
+
+We use plain custom elements. This does not stop us from depending on advanced base classes, but we get a lot of mileage out of the supported API now.
+
+### Applying CSS only to instantiated elements
+
+We can use the :defined CSS selector to apply styling rules only when the custom element&rsquo;s JavaScript has run. We import and deliver this CSS as part of the JavaScript package because it&rsquo;s only used in the js-powered version of the component
+
```

Comment:
- I'm not sure I understand this one intuitively. I think we should reference an example. 
- Add a period (.) to the end of the sentence. 
- "it is" instead of "it's"

Reply:
How about this:

We can use the ```:defined``` CSS selector to apply styling rules only when the custom element&rsquo;s JavaScript has run. 

For example we reference the custom element inside our CSS file with and without ```:defined```:

```
cagov-accordion {
	/* these style rules are applied always */
}
```

```
cagov-accordion:defined {
	/* these style rules are applied only after the JavaScript code associated with this element has been run */
}
```

We import and deliver this CSS as part of the JavaScript package because it&rsquo;s only used in the js-powered version of the component

Reply:
updated